### PR TITLE
Pruning fixes

### PR DESCRIPTION
**Pull request by mkaglins**
Fixes for gradients and BatchNorms zeroing (now all batch norms are masked with the previous convolution by default).

cc @AlexKoff88  @vanyalzr @AlexanderDokuchaev

### CODE REVIEW

**Comment by vanyalzr** [comment]
Does this diff affect the final accuracy of pruned models in any significant way?

**Comment by mkaglins** [comment]
> Does this diff affect the final accuracy of pruned models in any significant way?

On CIFAR-100 significant changes were not found (resnet-18, resnet-50, mobilenet). Experiments on Imagenet will be conducted to get more accurate results.

**Review by AlexKoff88 — COMMENTED** [review]
I am ok with PR but CI should be fixed.

**Review by vshampor — APPROVED** [review]

**Comment by mkaglins** [comment]
Results for resnet-18 pruning on Imagenet (config `resnet18_pruning_geometric_median.json`):
 Model name | Orinal acc | master | with fixes |
|------------|------------|--------|------------|
| resnet18   |    69.76   |  68.69 |    68.92   |
